libs/server/Custom/CustomCommandManagerSession.cs

```diff
@@ -1,100 +1,190 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-using System.Diagnostics;
+using System;
+using System.Collections.Generic;
 using Garnet.common;
 
 namespace Garnet.server
 {
     /// <summary>
-    /// Server session for RESP protocol - basic commands are in this file
+    /// Session-specific access to custom command data held by CustomCommandManager
+    /// For both custom procedures and custom transactions, this class maintains a cached instance of the transaction / procedure that can be called directly per-session
+    /// This class also maintains a cache of custom command info and docs to avoid the cost of synchronization when calling CustomCommandManager
     /// </summary>
     internal sealed class CustomCommandManagerSession
     {
+        // Initial size of expandable maps
+        private static readonly int MinMapSize = 8;
```

Comment:
why not start the expandable map at 0 so that we dont pay the cost of memory in the common case that there are no custom commands? then then grow as 1,2,4, etc.